x/poolmanager/keeper/create_pool.go

```diff
@@ -61,11 +73,6 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg types.CreatePoolMsg) (uint64, er
 		return 0, err
 	}
 
-	// create and save the pool's module account to the account keeper
-	if err := osmoutils.CreateModuleAccount(ctx, k.accountKeeper, pool.GetAddress()); err != nil {
```

Comment:
@omritoptix this method was used to create module accounts. It set's name == address, which is wrong for Module accounts. (as module account have name-> Addr known transformation)

Reply:
I set the Name and Addr to be `poolID->Name->Addr`
instead of 'poolID->Addr`